# HG changeset patch
# User Kai T. Ohlhus <k.ohlhus@gmail.com>
# Date 1521122463 -3600
#      Thu Mar 15 15:01:03 2018 +0100
# Branch stable
# Node ID 088506eadb44ecade8c0d81aa2f1f04e64af5801
# Parent  655dfe391ec2ef95cad5799040cde4d46d808f62
Enable build of a reduced Doxygen version for compressed Qt help file.

Currently a doxygen build is about 2 GB, certainly too much for any file
distribution.  Thus `make doxyqt` now creates a complete Doxygen build of about
70 MB without dot graphics and Javascript search tools.

* doc/doxyhtml/Doxyfile.in: Update year.  Build Javascipt search engine by
default HTML generation.

* doc/doxyhtml/DoxyfileQt.patch: A patch for Doxyfile.in to make the reduced
build for compressed Qt help files.

* doc/doxyhtml/module.mk: Add target `doxyqt` for reduced build in subfolder
`doxyhtml/doxyqt`.  Add new files to build system.

diff --git a/doc/doxyhtml/Doxyfile.in b/doc/doxyhtml/Doxyfile.in
--- a/doc/doxyhtml/Doxyfile.in
+++ b/doc/doxyhtml/Doxyfile.in
@@ -1,11 +1,11 @@
 # -*- mode: conf; -*-
 
-## Copyright (C) 2013-2017 Jordi Gutiérrez Hermoso
+## Copyright (C) 2013-2018 Jordi Gutiérrez Hermoso
 ##
 ## This file is part of Octave.
 ##
 ## Octave is free software: you can redistribute it and/or modify it
 ## under the terms of the GNU General Public License as published by
 ## the Free Software Foundation, either version 3 of the License, or
 ## (at your option) any later version.
 ##
@@ -575,16 +575,37 @@ DISABLE_INDEX          = YES
 # the root of the tree instead of the PROJECT_NAME. Since the tree basically has
 # the same information as the tab index, you could consider setting
 # DISABLE_INDEX to YES when enabling this option.
 # The default value is: NO.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
 GENERATE_TREEVIEW      = YES
 
+# When the SEARCHENGINE tag is enabled doxygen will generate a search box for
+# the HTML output. The underlying search engine uses javascript and DHTML and
+# should work on any modern browser. Note that when using HTML help
+# (GENERATE_HTMLHELP), Qt help (GENERATE_QHP), or docsets (GENERATE_DOCSET)
+# there is already a search function so this one should typically be disabled.
+# For large projects the javascript based search engine can be slow, then
+# enabling SERVER_BASED_SEARCH may provide a better solution. It is possible to
+# search using the keyboard; to jump to the search box use <access key> + S
+# (what the <access key> is depends on the OS and browser, but it is typically
+# <CTRL>, <ALT>/<option>, or both). Inside the search box use the <cursor down
+# key> to jump into the search results window, the results can be navigated
+# using the <cursor keys>. Press <Enter> to select an item or <escape> to cancel
+# the search. The filter options can be selected when the cursor is inside the
+# search box by pressing <Shift>+<cursor down>. Also here use the <cursor keys>
+# to select a filter and <Enter> or <escape> to activate or cancel the filter
+# option.
+# The default value is: YES.
+# This tag requires that the tag GENERATE_HTML is set to YES.
+
+SEARCHENGINE           = YES
+
 #---------------------------------------------------------------------------
 # Configuration options related to the LaTeX output
 #---------------------------------------------------------------------------
 
 # If the GENERATE_LATEX tag is set to YES, doxygen will generate LaTeX output.
 # The default value is: YES.
 
 GENERATE_LATEX         = NO
diff --git a/doc/doxyhtml/DoxyfileQt.patch b/doc/doxyhtml/DoxyfileQt.patch
new file mode 100644
--- /dev/null
+++ b/doc/doxyhtml/DoxyfileQt.patch
@@ -0,0 +1,8 @@
+
+# The following lines are used to create a minimal version to be used
+# as documentation within the GNU Octave GUI widget Qt Assistant.
+
+HTML_OUTPUT            = doxyhtml/doxyqt
+HAVE_DOT               = NO
+GENERATE_TREEVIEW      = NO
+SEARCHENGINE           = NO
diff --git a/doc/doxyhtml/module.mk b/doc/doxyhtml/module.mk
--- a/doc/doxyhtml/module.mk
+++ b/doc/doxyhtml/module.mk
@@ -3,21 +3,28 @@
 %reldir%/pages/README.md: $(srcdir)/README
 	$(MKDIR_P) $(@D)
 	cat $< | $(SED) '1s/.*/notitle {#mainpage}/; 2s/.*/=======/' > $@
 
 DOXYGEN_PAGES = \
   %reldir%/pages/macros.dox \
   %reldir%/pages/README.md
 
+doxyqt: %reldir%/Doxyfile $(DOXYGEN_PAGES) | %reldir%/$(octave_dirstamp)
+	cp %reldir%/Doxyfile %reldir%/DoxyfileQt
+	cat $(srcdir)/%reldir%/DoxyfileQt.patch >> %reldir%/DoxyfileQt
+	doxygen %reldir%/DoxyfileQt
+
 doxyhtml: %reldir%/Doxyfile $(DOXYGEN_PAGES) | %reldir%/$(octave_dirstamp)
 	doxygen %reldir%/Doxyfile
 
+# This target is important for in source tree builds.
 doxyhtml-maintainer-clean:
-	rm -f %reldir%/pages/README.md
-	rm -rf `ls -d %reldir%/* 2>/dev/null | $(GREP) -v 'module\.mk\|Doxyfile\.in\|README\|pages$$'`
+	$(RM) %reldir%/pages/README.md
+	$(RM) -r `ls -d %reldir%/* 2>/dev/null | $(GREP) -v 'module\.mk\|Doxyfile\.in\|DoxyfileQt\.patch\|README\|pages$$'`
 
 doc_EXTRA_DIST += \
   $(DOXYGEN_PAGES) \
   %reldir%/Doxyfile.in \
+  %reldir%/DoxyfileQt.patch \
   %reldir%/README
 
 DIRSTAMP_FILES += %reldir%/$(octave_dirstamp)
